test/extended/federation.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+#
+# Runs the federation tests against a standard started server.
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+source "${OS_ROOT}/test/extended/setup.sh"
+
+os::test::extended::setup
+
+OS_FEDERATION_NAME="${OS_FEDERATION_NAME:-origin-federation}"
+OS_FEDERATION_NAMESPACE="${OS_FEDERATION_NAMESPACE:-federation-system}"
+SERVICE_ACCOUNT="system:serviceaccount:${OS_FEDERATION_NAMESPACE}:default"
+
+os::log::info ""
+os::log::info "Deploying federation control plane"
+kubefed init "${OS_FEDERATION_NAME}" --dns-provider=google-clouddns \
+        --federation-system-namespace="${OS_FEDERATION_NAMESPACE}" \
+        --etcd-persistent-storage=false
+
+# Ensure the controller manager will be able to access cluster configuration stored as
+# secrets in the federation namespace.
+os::log::info ""
+os::log::info "Granting the admin role to the service account of the federation namespace"
+oadm --namespace "${OS_FEDERATION_NAMESPACE}" policy add-role-to-user admin "${SERVICE_ACCOUNT}"
+
+# TODO enable coredns when documentation is available
+os::log::info ""
+os::log::info "Disabling the federation services controller to avoid having to configure dnsaas"
+kubectl --namespace="${OS_FEDERATION_NAMESPACE}" patch deploy "${OS_FEDERATION_NAME}-controller-manager" \
+        --type=json -p='[{"op": "add", "path": "/spec/template/spec/containers/0/command/-", "value": "--controllers=services=false"}]'
+
+os::log::info ""
+os::log::info "Waiting for federation api to become available"
+os::cmd::try_until_text "oc get --raw /healthz --config='${ADMIN_KUBECONFIG}' --context=${OS_FEDERATION_NAME}" 'ok' $(( 80 * second )) 0.25
```

Comment:
Also suggest wrapping the other set-up commands in `os::cmd::expect_success` so that we can inspect set-up failures as well.